prog/weaver/main.go

```diff
@@ -273,10 +273,25 @@ func main() {
 		dnsserver *nameserver.DNSServer
 	)
 	if !noDNS {
-		ns, dnsserver = createDNSServer(dnsConfig, router.Router, isKnownPeer)
+		var existingContainerIDs []string
+
+		ns, dnsserver = createDNSServer(dnsConfig, router.Router, db, isKnownPeer)
 		observeContainers(ns)
-		ns.Start()
+
+		if dockerCli != nil {
+			existingContainerIDs, err = dockerCli.ExistingContainerIDs()
+			if err != nil {
+				Log.Fatalf("ExistingContainerIDs failed due to: %s", err)
+			}
+		}
+
+		// To restore external DNS entries we should pretend that "weave:extern"
+		// container exists.
+		existingContainerIDs = append(existingContainerIDs, "weave:extern")
+		ns.Start(existingContainerIDs)
+		ns.RestoreEntries("weave:extern")
```

Comment:
Yep, it's needed. `ns.restoreFromDB` (called by `ns.Start`) only flags entries of "weave:extern" as `stopped`, while `ns.RestoreEntries` does "un-tombstoning" of the entries.